newton/_src/solvers/mujoco/solver_mujoco.py

```
@@ -3077,7 +3078,33 @@ def add_geoms(newton_body_id: int):
                     "name": name,
                 }
                 tf = wp.transform(*shape_transform[shape])
-                if stype == GeoType.MESH or stype == GeoType.CONVEX_MESH:
+                if stype == GeoType.HFIELD:
+                    # Retrieve heightfield source
+                    hfield_src = model.shape_source[shape]
+                    if hfield_src is None:
+                        if verbose:
```

Ruff (0.14.14)

[error] Undefined name `verbose`

(F821)